### ISSUE TITLE
Chord error handling example in docs not working as expected

### ISSUE THREAD

**Issue by OskarPersson** [celery/celery#5129]
## Checklist

- [x] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

```

$ celery -A tasks report

No handlers could be found for logger "vagrant"

software -> celery:4.2.0 (windowlicker) kombu:4.2.1 py:2.7.14
            billiard:3.5.0.4 py-amqp:2.3.2
platform -> system:Linux arch:64bit imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:amqp results:redis://redis/

broker_url: u'amqp://guest:********@rabbit:5672//'
result_backend: u'redis://redis/'
```

## Steps to reproduce
Add `tasks.py`:

```python
from celery import Celery, chord

app = Celery('tasks',
             backend='redis://redis/',
             broker='amqp://guest:guest@rabbit:5672//')

@app.task
def add(x, y):
    return x + y

@app.task
def raising_task():
    raise ValueError('something something')


if __name__ == '__main__':
    c = chord([add.s(4, 4), raising_task.s(), add.s(8, 8)])
    result = c()
    result.get()
```
Run the tasks:

```
$ python tasks.py
```

## Expected behavior

According to the [documentation](http://docs.celeryproject.org/en/latest/userguide/canvas.html#error-handling) this should result in something like the following being raised on the worker:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "*/celery/result.py", line 120, in get
    interval=interval)
  File "*/celery/backends/amqp.py", line 150, in wait_for
    raise meta['result']
celery.exceptions.ChordError: Dependency 97de6f3f-ea67-4517-a21c-d867c61fcb47
    raised ValueError('something something',)
```

## Actual behavior

A traceback is raised before any task is sent to the workers

```
Traceback (most recent call last):
  File "tasks.py", line 18, in <module>
    result = c()
  File "/home/developer/celery/canvas.py", line 1189, in __call__
    return self.apply_async((), {'body': body} if body else {}, **options)
  File "/home/developer/celery/canvas.py", line 1224, in apply_async
    body = body.clone(**options)
AttributeError: 'NoneType' object has no attribute 'clone'
```

**Comment by OskarPersson** [comment]
Yes, that is probably the issue. Though how would you then run an **error callback** when a group fails without defining a callback task?

**Comment by xirdneh** [comment]
A chord is a chain with a group and another task `chain(group([...]) | callback.s())`.
And a chain and a chord are signature objects. So, you can use `on_error` to assign an error callback.
```python
>>> chord(header)(callback.s().on_error(error_cb.s()))
>>> chain(group | callback).on_error(error_cb.s())
```
You can also try to set `on_error` within the group:
```python
>>> g = group(t1.s(), t2.s(), link_error=error_cb.s())
>>> chord(g)(callback.s())
```
